### PR TITLE
Configure and fix linting to give good warnings

### PR DESCRIPTION
**Pull request by ssorallen**
* Enable necessary globals: jquery, webextensions, and moment
* Disable `console` use warnings since they are used throughout the codebase
* Run ESLint only on carettab code, not on vendor code unowned by this repo
* Autofix all auto-fixable warnings with `npm run-script lint -- --fix`

### CODE REVIEW

**Comment by ssorallen** [comment]
This does not yet make lint errors too useful because the files rely heavily on declaring and using globals. This is a step in the right direction though. You could continue using globals and declare them using ESLint's `/* global */` syntax or move to a module loader + bundler like [Webpack](https://webpack.js.org/).

File: .eslintrc.json [hunk 1]
```diff
@@ -12,6 +17,9 @@
       "error",
       "windows"
     ],
+    "no-console": [
+      "off"
+    ],
     "quotes": [
       "error",
       "double"
```
Comment (bluecaret) [inline .eslintrc.json]: Any particular reason for the preference of double quotes? I usually prefer single quotes myself and is what I typically see being used.

Comment (ssorallen) [reply .eslintrc.json]: .eslintrc.json was already in your project! I didn't change the `"quotes"` preference, I assumed this file had the settings you wanted.

I prefer single quotes too. I can change that.

**Comment by ssorallen** [comment]
Would you like to use this pull request? Since you're rewriting the project, I'll close this if you have other changes already.

File: .eslintrc.json (hunk 1 above)
Comment (bluecaret) [reply .eslintrc.json]: Look at that I did lol. I think when I originally built this extension I looked for a basic setup, and it must have included that, just never used it yet :)

**Comment by bluecaret** [comment]
@ssorallen There will be big changes including a different lint file so ya lets just close this for now. Thanks for the work though!

**Pull request closed without merging**